text.go

```diff
@@ -0,0 +1,236 @@
+	return self.text.Font
+}
+
+// Sets canvas' default font family
+func (self *Canvas) SetFontFamily(family string) {
+	self.text.Family = family
+	cfamily := C.CString(family)
+	defer C.free(unsafe.Pointer(cfamily))
+	C.DrawSetFontFamily(self.drawing, cfamily)	
+}
+
+// Returns canvas' current font family
+func (self *Canvas) FontFamily() string {
+	return self.text.Family
+}
+
+// Sets canvas' default font size
+func (self *Canvas) SetFontSize(size float64) {
+	self.text.Size = size
+	C.DrawSetFontSize(self.drawing, C.double(size))		
+}
+
+// Returns canvas' current font size
+func (self *Canvas) FontSize() float64 {
+	return self.text.Size
+}
+
+// Sets canvas' default font weight
+func (self *Canvas) SetFontWeight(weight uint) {
+	self.text.Weight = weight
+	C.DrawSetFontWeight(self.drawing, C.size_t(weight))		
+}
+
+// Returns canvas' current font weight
+func (self *Canvas) FontWeight() uint {
+	return self.text.Weight
+}
+
+
+// Sets canvas' font name and size.
+// If font is 0-length, the current font family is not changed
+// If size is <= 0, the current font size is not changed
+func (self *Canvas) SetFont(font string, size float64) {
+	if len(font) > 0 {
+		self.SetFontName(font)
+	}
+	if size > 0 {
+		self.SetFontSize(size)
+	}
+}
+
+// Returns canvas' current font name and size
+func (self *Canvas) Font() (string, float64) {
+	return self.text.Font, self.text.Size
+}
+
+// Sets canvas' default text alignment. Available values are:
+// UndefinedAlign (?), LeftAlign, CenterAlign, RightAlign
+func (self *Canvas) SetTextAlignment(a Alignment) {
+	self.text.Alignment = a
+	C.DrawSetTextAlignment(self.drawing, C.AlignType(a))
+}
+
+// Returns the canvas' current text aligment
+func (self *Canvas) TextAlignment() Alignment {
+	return self.text.Alignment
+}
+
+// Sets canvas' default text antialiasing option.
+func (self *Canvas) SetTextAntialias(b bool) {
+	self.text.Antialias = b
+	C.DrawSetTextAntialias(self.drawing, magickBoolean(b))
+}
+
+// Returns the canvas' current text aligment
+func (self *Canvas) TextAntialias() bool {
+	return self.text.Antialias
+}
+
+// Sets canvas' default text antialiasing option.
+func (self *Canvas) SetTextKerning(k float64) {
+	self.text.Kerning = k
+	C.DrawSetTextKerning(self.drawing, C.double(k))
+}
+
+// Returns the canvas' current text aligment
+func (self *Canvas) TextKerning() float64 {
+	return self.text.Kerning
+}
+
+// Draws a string at the specified coordinates and using the current canvas
+// Alignment.
+func (self *Canvas) Annotate(text string, x, y float64) {
+	c_text := C.CString(text)
+	defer C.free(unsafe.Pointer(c_text))
+	C.DrawAnnotation(self.drawing, C.double(x), C.double(y), (*C.uchar)(unsafe.Pointer(c_text)))
+}
+
+// Draws a string at the specified coordinates and using the specified Text Properties
+// Does not modify the canvas' default TextProperties
+func (self *Canvas) AnnotateWithProperties(text string, x, y float64, prop *TextProperties) {
+	if prop != nil {
```

Comment:
The if is only to avoid useless operations : copying the current `TextProperties`, calling `SetProperties` with `nil` then calling it again with the copied `TextProperties`.
I'd keep it that way only for performances. Don't you agree @phacops?